docusaurus/docs/protocol/upgrades/upgrade_procedure.md

```diff
@@ -91,18 +141,89 @@ Note that for local testing, `cosmovisor` won't pull the binary from the info fi
+LocalNet **DOES NOT** support `cosmovisor` and automatic upgrades at the moment.
+
+However, **IT IS NOT NEEDED** to simulate and test the upgrade procedure.
+
+#### LocalNet Upgrade tl;dr
+
+1. Pull git repo with old version (separate directory)
+2. Download release binary of the old version
+3. Wipe LocalNet data and generate genesis using OLD version
+4. Start node using anOLD binary
+5. Write and submit an upgrade transaction on-chain
+6. When the Upgrade Plan height is reached, stop the old node and run the new binary
+7. Observe the behavior
+
+#### LocalNet Upgrade Full Example Walkthrough
+
+Testing an upgrade requires a network running on an old version.
+
+Ensure LocalNet is running using a binary from the [previous release you wish to upgrade **FROM**](https://github.com/pokt-network/poktroll/releases). We also want to provision the network using this version, which requires us to pull the specific git tag.
+
+1. Make a note of the version you want to test an upgrade **FROM**. This will be the **OLD** version. For example, let's imagine we're upgrading from `v0.0.9`.
+2. Pull a new `poktroll` repo (will be used as an "old" version):
+
+   ```bash
+   git clone https://github.com/pokt-network/poktroll.git poktroll-upgrade-old
+   cd poktroll-upgrade-old
+   git checkout v0.0.9
+
+   # Download the v0.0.9 binary: https://github.com/pokt-network/poktroll/releases
+   # CHANGE POKTROLLD_VERSION and ARCH
+   curl -L "https://github.com/pokt-network/poktroll/releases/download/${POKTROLLD_VERSION}/poktroll_linux_${ARCH}.tar.gz" | tar -zxvf - -C .
+
+   # Validate the version
+   ./poktrolld version
+   0.0.9
+   ```
+
+3. Stop LocalNet
+
+   ```bash
+   make localnet_down
+   ```
+
+4. Reset the data
+
+   ```bash
+   ./poktrolld comet unsafe-reset-all
+   ```
+
+5. Create new genesis using old version (from `poktroll-upgrade-old` dir)
+
+   ```bash
+   make localnet_regenesis
+   ```
+
+6. Start the network
+
+   ```bash
+   ./poktrolld start
+   ```
+
+7. [Write](#writing-an-upgrade-transaction) and [Submit](#submitting-the-upgrade-on-chain) a transaction. For example:
+
+   ```bash
+   poktrolld tx authz exec tools/scripts/upgrades/local_test_v0.0.9-2.json --from=pnf`
+   ```
 
-1. Modify `tools/scripts/upgrades/authz_upgrade_tx_example_v0.0.4_height_30.json` to reflect the name of the upgrade and the height at which it should be scheduled.
+8. Verify the plan is active
 
-2. Check and update the `tools/scripts/upgrades/cosmovisor-start-node.sh` to point to the correct binaries:
+   ```bash
+   poktrolld query upgrade plan
+   ```
 
-   - The old binary should be compiled to work before the upgrade.
-   - The new binary should contain the upgrade logic to be executed immediately after the node is started using the new binary.
+9. Wait until the height is reached and the old node dies due to the error: `ERR UPGRADE "v0.0.9-2" NEEDED at height`, which is expected.
```

Comment:
I did a little bit of partial cleanup, but can you do another look through this and make it so copy-pastable that any idiot can follow?

For example, instead of "from the new version", just add the appropriate `cd ../...` in there.

Reply:
How do I know what the new version path on the developer's system? It might be a primary `poktroll` repo on the developer machine, it might not be. ¯\\\_(ツ)_/¯

Reply:
Maybe an instruction where they run `poktroll version` to ensure it's correct?

It's one of the steps in the upgrade playbook for Morse:

<img width="878" alt="Screenshot 2024-10-24 at 12 18 01 PM" src="https://github.com/user-attachments/assets/0cc88b30-5c98-401f-a48c-f346b60ea1c2">

Reply:
Started by writing a cheat sheet, ended up changing the whole section.